Sort a work's files in alphabetical order by default

nicktiberi opened the pull request:
Fixes #1044.

nicktiberi committed:
Sort a work's files in alphabetical order by default

awead commented on spec/models/work_version_spec.rb:
```diff
@@ -59,6 +59,14 @@
+      let(:resource) { create(:work_version, :with_files, file_count: 2) }
+
+      it 'orders the files by title in ascending order' do
+        expect(resource.file_version_memberships.map(&:title)).to eq(['image-1.png', 'image-2.png'])
```
You need to rename the files and verify that they get sorted correctly. Something like:
``` ruby
resource.file_version_memberships[0].update(title: 'image-4.png')
resource.file_version_memberships[1].update(title: 'image-3.png')
```
ought to do the trick.

awead commented on app/models/file_version_membership.rb:
```diff
@@ -4,6 +4,8 @@ class FileVersionMembership < ApplicationRecord
   belongs_to :work_version
   belongs_to :file_resource
 
+  default_scope { order(title: :asc) }
```
ah nice! I forgot about that....

nicktiberi commented:
@awead - I have one failing test and am wondering if you have any idea why it's failing. 

I ran `spec/services/file_version_membership_diff_spec.rb` locally and everything passed, but it seems to still be failing in CI.

Based on expected/got, it seems like the timestamps are different, and that may be what's causing it to fail. But I'm not sure why the timestamps are different.

```
Failures:

  1) FileVersionMembershipDiff when a file has been renamed is expected to eq {:added=>[], :deleted=>[], :renamed=>[[#<FileVersionMembership id: 537, work_version_id: 1014, file_r...ted_at: "2021-06-01 11:01:39.599640000 -0400", updated_at: "2021-06-01 11:01:39.599640000 -0400">]]}
     Failure/Error: it { is_expected.to eq(renamed: [[original_second_image, updated_second_image]], added: [], deleted: []) }
     
       expected: {:added=>[], :deleted=>[], :renamed=>[[#<FileVersionMembership id: 537, work_version_id: 1014, file_r...ted_at: "2021-06-01 11:01:39.599640000 -0400", updated_at: "2021-06-01 11:01:39.599640000 -0400">]]}
            got: {:added=>[], :deleted=>[], :renamed=>[[#<FileVersionMembership id: 537, work_version_id: 1014, file_r...ted_at: "2021-06-01 11:01:39.602285000 -0400", updated_at: "2021-06-01 11:01:39.611688000 -0400">]]}
     
       (compared using ==)
     
       Diff:
       @@ -1,4 +1,4 @@
        :added => [],
        :deleted => [],
       -:renamed => [[#<FileVersionMembership id: 537, work_version_id: 1014, file_resource_id: 546, title: "image-554.png", created_at: "2021-06-01 11:01:39.566987000 -0400", updated_at: "2021-06-01 11:01:39.566987000 -0400">, #<FileVersionMembership id: 538, work_version_id: 1015, file_resource_id: 545, title: "image-553.png", created_at: "2021-06-01 11:01:39.599640000 -0400", updated_at: "2021-06-01 11:01:39.599640000 -0400">]],
       +:renamed => [[#<FileVersionMembership id: 537, work_version_id: 1014, file_resource_id: 546, title: "image-554.png", created_at: "2021-06-01 11:01:39.566987000 -0400", updated_at: "2021-06-01 11:01:39.566987000 -0400">, #<FileVersionMembership id: 539, work_version_id: 1015, file_resource_id: 546, title: "Image 2.png", created_at: "2021-06-01 11:01:39.602285000 -0400", updated_at: "2021-06-01 11:01:39.611688000 -0400">]],
       
     # ./spec/services/file_version_membership_diff_spec.rb:22:in `block (3 levels) in <top (required)>'

Finished in 7 minutes 10 seconds (files took 11.19 seconds to load)
2092 examples, 1 failure, 20 pending

Failed examples:

rspec ./spec/services/file_version_membership_diff_spec.rb:22
```

nicktiberi commented:
@awead - I think codeclimate is just stuck. This should be good for re-review now.

awead commented on spec/models/work_version_spec.rb:
```diff
@@ -59,6 +59,24 @@
+
+      it 'orders the files by title in ascending order' do
+        n1 = (resource.file_version_memberships[0].title.match /image-(\d+).png/)[1].to_i
+        n2 = n1 + 1
```
clever!
